File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/dashboard/DashBoardFragment.java
package com.justnik.mirea_navigation_lesson.ui.dashboard;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.justnik.mirea_navigation_lesson.R;
import com.justnik.mirea_navigation_lesson.databinding.FragmentDashBoardBinding;
import com.justnik.mirea_navigation_lesson.enteties.City;

public class DashBoardFragment extends Fragment {
    private final static String TYUMEN_IMAGE_URL = "https://img4.teletype.in/files/71/cd/71cde14e-9f12-4970-9dab-f16c7048edc5.jpeg";
    private final static String MOSCOW_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/db/View_of_Moscow1.jpg/481px-View_of_Moscow1.jpg";

    private FragmentDashBoardBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentDashBoardBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        setClickListeners();
    }

    private void setClickListeners() {
        binding.bTyumen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                City city = createCity("Тюмень(Лучший город земли)", 816_700, TYUMEN_IMAGE_URL);
                Bundle bundle = CityFragment.createBundle(city);
                Navigation.findNavController(view).navigate(
                        R.id.action_dashBoardFragment_to_cityFragment,
                        bundle
                );

            }
        });

        binding.bMoscow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                City city = createCity("Москва", 12_635_466, MOSCOW_IMAGE_URL);
                Bundle bundle = CityFragment.createBundle(city);
                Navigation.findNavController(view).navigate(
                        R.id.action_dashBoardFragment_to_cityFragment,
                        bundle
                );

            }
        });
    }

    private City createCity(String name, int population, String imageUrl) {
        return new City(name, population, imageUrl);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }


}

File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/dashboard/DashBoardFragment.kt
package com.justnik.mirea_navigation_lesson.ui.dashboard

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import by.kirich1409.viewbindingdelegate.viewBinding
import com.justnik.mirea_navigation_lesson.R
import com.justnik.mirea_navigation_lesson.databinding.FragmentDashBoardBinding
import com.justnik.mirea_navigation_lesson.enteties.City

class DashBoardFragment : Fragment(R.layout.fragment_dash_board) {

    private val binding: FragmentDashBoardBinding by viewBinding()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        setClickListeners()
    }

    private fun setClickListeners() {
        binding.bTyumen.setOnClickListener {
            val city = createCity("Тюмень(Лучший город земли)", 816_700, TYUMEN_IMAGE_URL)
            findNavController().navigate(
                DashBoardFragmentDirections.actionDashBoardFragmentToCityFragment(city)
            )

        }
        binding.bMoscow.setOnClickListener {
            val city = createCity("Москва", 12_635_466, MOSCOW_IMAGE_URL)
            findNavController().navigate(
                DashBoardFragmentDirections.actionDashBoardFragmentToCityFragment(city)
            )
        }
    }


    private fun createCity(name: String, population: Int, imageUrl: String): City {
        return City(
            name = name,
            population = population,
            imageUrl = imageUrl
        )
    }

    companion object{
        private const val TYUMEN_IMAGE_URL = "https://img4.teletype.in/files/71/cd/71cde14e-9f12-4970-9dab-f16c7048edc5.jpeg"
        private const val MOSCOW_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/db/View_of_Moscow1.jpg/481px-View_of_Moscow1.jpg"
    }
}